src/higher-order.js

```diff
@@ -152,6 +153,11 @@ function branch(cursors, Component) {
       this.watcher.refresh(mapping);
       this.setState(this.watcher.get());
     }
+
+    // Should update the component?
+    shouldComponentUpdate(nextProps, nextState) {
+      return !(deepEqual(this.props, nextProps, {strict: true}) && deepEqual(this.state, nextState, {strict: true}))
```

Comment:
I got following numbers:
<img width="309" alt="screen shot 2017-03-24 at 3 37 13 pm" src="https://cloud.githubusercontent.com/assets/5691667/24315991/ea6858b8-10a7-11e7-899d-fdb3dd2343cc.png">

`hybrid` is using deepEqual per prop.
Given that first line of code in deepEqual is `===` check, it takes no extra time overall than `===`(for some reason it took even less time in experiment:)) but it prevents 8 re-renders which can be worth a lot of cpu time.
It'll also prevent app getting slow if we are doing tree.set(sameObjectWithNewReference) in the app which we seem to given these 8 extra re-renders, which ideally we should investigate separately and fix the root cause.

So I am going ahead with using deepEqual per property.